contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/rule_based_profiler/domain_builder/data_profiler_column_domain_builder.py

```diff
@@ -0,0 +1,147 @@
+        ] = None,
+        exclude_semantic_types: Optional[
+            Union[str, SemanticDomainTypes, List[Union[str, SemanticDomainTypes]]]
+        ] = None,
+        data_context: Optional[AbstractDataContext] = None,
+    ) -> None:
+        super().__init__(
+            include_column_names=include_column_names,
+            exclude_column_names=exclude_column_names,
+            include_column_name_suffixes=include_column_name_suffixes,
+            exclude_column_name_suffixes=exclude_column_name_suffixes,
+            semantic_type_filter_module_name=semantic_type_filter_module_name,
+            semantic_type_filter_class_name=semantic_type_filter_class_name,
+            include_semantic_types=include_semantic_types,
+            exclude_semantic_types=exclude_semantic_types,
+            data_context=data_context,
+        )
+
+    def _get_domains(
+        self,
+        rule_name: str,
+        variables: Optional[ParameterContainer] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> List[Domain]:
+        """
+        Obtains and returns domains for all columns of a table that match a rule.
+        Args:
+            rule_name: name of Rule object, for which "Domain" objects are obtained.
+            variables: Optional variables to substitute when evaluating.
+            runtime_configuration: Optional[dict] = None,
+        Returns:
+            List of domains that match the desired columns and filtering criteria.
+        """
+        batch_ids: List[str] = self.get_batch_ids(
+            variables=variables
+        )  # type: ignore[assignment] # could be None
+
+        # type: ignore[assignment] # could be None
+        validator: Validator = self.get_validator(variables=variables)
+
+        effective_column_names: List[str] = self.get_effective_column_names(
+            batch_ids=batch_ids,
+            validator=validator,
+            variables=variables,
+            rule_name=rule_name,
+        )
+
+        column_name: str
+        domains: List[Domain] = build_domains_from_column_names(
+            rule_name=rule_name,
+            column_names=effective_column_names,
+            domain_type=self.domain_type,
+            # type: ignore[union-attr] # could be None
+            table_column_name_to_inferred_semantic_domain_type_map=None,
+            # self.semantic_type_filter.table_column_name_to_inferred_semantic_domain_type_map,
+        )
+
+        return domains
+
+    def get_effective_column_names(
+        self,
+        batch_ids: Optional[List[str]] = None,
+        validator: Optional[Validator] = None,
+        variables: Optional[ParameterContainer] = None,
+        rule_name: Optional[str] = None,
+    ) -> List[str]:
+        """
+        This method filters by a rule's criteria to obtain columns to be included as part of returned "Domain" objects.
+        """
+
+        effective_column_names = list()
+
+        rule_name_to_data_types = {
```

Comment:
@alexjeilert A shortcoming of this approach is that the keys in this table are entirely arbitrary.  Perhaps we could come up with a more robust alternative?  Towards this goal, could you please provide a paragraph or two description of the goals that this new "column-kind" `DomainBuilder` should aim to achieve?  Thank you very much (please also feel free to contact me on Slack).